### PR TITLE
Add clusterplex Helm chart

### PR DESCRIPTION
**Pull request by brandan-schmitz**
### Overview
This pull requests adds a helm chart for ClusterPlex as was originally discussed in #156 last year. I have finally gotten around to creating the helm chart as I deployed it on my new Kubernetes cluster.

This helm chart uses the [common library chart](https://github.com/bjw-s/helm-charts/tree/main/charts/library/common) created by [bjw-s](https://github.com/bjw-s) as a foundation and simply configures it to deploy ClusterPlex. This library chart is a next-generation version of the popular, yet now defunct k8s-at-home helm chart.

<br>

### Workflow File
You will notice in this pull request I included a GitHub Actions workflow file that will do the following:

1. Install necessary dependencies to package and release the chart such as helm  and helm-docs used to generate the charts README file.
2. Generate the helm docs and utilize the latest README as the main page that will be displayed on the GitHub pages page for the helm chart repo.
3. Package the helm chart.
4. Save the new helm chart package, updated README and update index file to the gh-pages branch.
5. Wait for the gh-pages deployment to be completed.

The above workflow is configured to run anytime a push is made to the `master` branch or a tag is created such as upon a release. It is worth noting that if a change is made to a file in the helm chart but the version of the chart is not updated, then it will end up overwriting the currently released version of the chart. Alternatively it can be changed to only be run upon a tag being pushed, perhaps even something like `chart-X.X.X` ensuring that the chart is only updated when needed. That can be changed by modifying the `on` section of the chart.yaml workflow file. If you want this done before merging let me know.

Using this workflow file and the below steps for setting up the GitHub Pages page for this repo will allow people to use the following helm command to add it as a repo containing the helm chart and to install it.

```shell
helm repo add clusterplex https://pabloromeo.github.io/clusterplex
helm upgrade --install clusterplex clusterplex/clusterplex
```

<br>

### Chart Versioning

You will notice in the `Chart.yaml` file it does show the `appVersion`. Upon a release of ClusterPlex, an updated helm chart should be released with the `appVersion` getting updated as well. Any new release of the chart should also increment the `version` of the chart. I did not tie the version of the chart to the ClusterPlex version in order to allow updates to the chart to be made separately for easier maintenance and releases.

Usually when I deal with chart versioning, a patch level version change in the application will be a patch level version change in the chart, same with minor and major level version changes. In addition, if I make a small patch-type change to the chart I increment just the chart version leaving the appVersion the same. If I were to update the common chart dependency by a patch version then I would also increase the chart version by a patch version, same with a minor and major change level. That is a big reason I usually do not tie a chart version to the app version.

<br>

### IMPORTANT: Before Merging
Before merging this pull request, you will want to have configured the repository to have a working GitHub pages page. You can do that by creating a new branch called `gh-pages` that should be orphaned from any other branch. That can be done on your computer in the clusterplex repo with the following commands:

```shell
git checkout --orphan gh-pages
git rm -rf .
git commit -m "Initial commit" --allow-empty
git push --set-upstream origin gh-pages
```

After you have created and pushed the empty `gh-pages` branch, you will need to go to your repo settings for this repo and under the **_Code and automation_** section in the left navigation menu select the _**Pages**_ option. Then under the source dropdown ensure the `Deploy from a branch` option is selected then under the Branch section select the `gh-pages` branch that was just created and for the folder ensure that it is set to `/ (root)` and click save.

### CODE REVIEW

**Comment by brandan-schmitz** [comment]
@pabloromeo Just wanted to bump ya and see if you have had a chance to review this or if you need any changes?

**Comment by pabloromeo** [comment]
Hi @brandan-schmitz! Sorry about the delay. I was just today going through it a bit.
I'm gonna set up the gh-pages stuff later today, to get ready to merge.
I haven't built my own helm charts before so I'll take your guidance on what's best, regarding package versioning and all that.
I'll post any comments on the PR once I'm done taking a look.

File: charts/clusterplex/templates/pms.yaml [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+      TZ: '{{ .Values.global.timezone | default "America/Chicago" }}'
+      PGID: '{{ .Values.global.PGID | default "1000" }}'
+      PUID: '{{ .Values.global.PUID | default "1000" }}'
+      DOCKER_MODS: 'ghcr.io/pabloromeo/clusterplex_dockermod:{{ .Values.global.clusterplexVersion | default .Chart.AppVersion }}'
+      ORCHESTRATOR_URL: 'http://{{ .Release.Name }}-orchestrator:{{ .Values.orchestrator.config.port | default "3500" }}'
+      PMS_SERVICE: '{{ .Release.Name }}-pms'
+      PMS_PORT: '{{ .Values.pms.config.port | default "32400" }}'
+      TRANSCODER_VERBOSE: '{{ .Values.pms.config.transcoderVerbose | default "1" }}'
+      TRANSCODE_OPERATING_MODE: '{{ .Values.pms.config.transcodeOperatingMode | default "both" }}'
+      LOCAL_RELAY_ENABLED: '1'
```
Comment (pabloromeo) [inline charts/clusterplex/templates/pms.yaml:24]: Any reason why disabling the Relay is not available through values.yaml? Just curious

**Review by pabloromeo — APPROVED** [review]
Looks great!

File: charts/clusterplex/templates/pms.yaml (hunk 1 above)
Comment (pabloromeo) [reply charts/clusterplex/templates/pms.yaml:24]: Let me know if you want me to merge as-is and we add that additional parameter at a later time, or if you want me to wait for an update. Don't think many will want to be turning that off anyways.

Comment (brandan-schmitz) [reply charts/clusterplex/templates/pms.yaml:24]: I originally did not add that in to keep it simple for the initial version. If you want to merge it in I can create a new PR adding the ability to change that in another version.

Comment (pabloromeo) [reply charts/clusterplex/templates/pms.yaml:24]: Sure thing, let's leave it as is.

**Comment by pabloromeo** [comment]
gh-pages already deployed https://pabloromeo.github.io/clusterplex/ 🍾 
Thank you so much for the awesome contribution @brandan-schmitz !!

**Pull request merged**